mdsuite/visualizer/visualizer.py

```diff
@@ -0,0 +1,229 @@
+"""
+This program and the accompanying materials are made available under the terms of the
+Eclipse Public License v2.0 which accompanies this distribution, and is available at
+https://www.eclipse.org/legal/epl-v20.html
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincware Project.
+
+Description: Visualize a simulation.
+"""
+from mdsuite.database.simulation_database import Database
+import open3d as o3d
+from os.path import join
+import tensorflow as tf
+import threading
+from PIL.ImageColor import getcolor
+import importlib.resources
+import json
+import numpy as np
+from mdsuite.utils.meta_functions import join_path
+
+
+class SimulationVisualizer:
+    """
+    Visualize a simulation.
+    """
+
+    def __init__(self,
+                 experiment: object,
+                 species: list = None,
+                 molecules: bool = False,
+                 unwrapped: bool = False):
+        """
+        Constructor for the visualizer.
+
+        Parameters
+        ----------
+        experiment : object
+                 Experiment object from which the visualizer will gather
+                 information.
+        species : list
+                A list of species to visualize.
+        molecules : list
+                If true, molecules will be visualized.
+        unwrapped : bool
+                If true, unwrapped coordinates are studied.
+        """
+        # Particle information
+        self.experiment = experiment
+        self.database = Database(name=join(self.experiment.database_path,
+                                           "database.hdf5"))
+        self.molecules = molecules
+        self.species = species
+        if unwrapped:
+            self.identifier = "Unwrapped_Positions"
+        else:
+            self.identifier = "Positions"
+        self._check_input()  # check inputs and build data structure.
+        self._fill_species_properties()
+
+        # App information
+        self.is_done = False
+        self.main_vis = None
+        self.snapshot_pos = None
+        self.n_snapshots = 0
+        self.mesh_dict = {}
+
+    def _fill_species_properties(self):
+        """
+        Add colours and sizes to species.
+
+        Returns
+        -------
+        Updates the class attributes.
+        """
+        data_name = "mdsuite.data"
+        json_name = 'PubChemElements_all.json'
+        with importlib.resources.open_text(data_name, json_name) as json_file:
```

Comment:
I just curious, are there any benefits over a plain `with open()` or `Path(data_name, json_name).read_text()` ?

Reply:
i didn't write any of this stuff so have no comment. I think it was Marco.

Reply:
Probably just another way to write it. It works, so resolved.